pyHalo/Halos/HaloModels/ULDM.py

```diff
@@ -86,9 +90,56 @@ def profile_args(self):
             truncation_radius = self._lens_cosmo.LOS_truncation_rN(self.mass, self.z,
                                                              self._args['LOS_truncation_factor'])
 
-            self._profile_args = (self.c, truncation_radius)
+            # using 'mass' as the ULDM virial mass
+            [theta_c, kappa_0] = self._uldm_args(self._args['m_uldm'], self.mass, self._args['uldm_plaw'])
+
+            self._profile_args = (self.c, truncation_radius, theta_c, kappa_0)
 
         return self._profile_args
+    
+    def _uldm_args(self, m, M, plaw):
+        """
+        Returns core radius and density in arcsec units. See eqns. (3) and (7) for the core radius
+        and core density, respectively, from Schive et al. 2014 [1407.7762v2].
+        """
+        m_log10 = np.log10(m)
+        M_log10 = np.log10(M)
+
+        a = 1/(1+self.z)
+        m22 = 10**(m_log10 + 22)
+        M9 = 10**(M_log10 - 9)
+
+        Sigma_crit = self._lens_cosmo.sigma_crit_lensing * 1e-12
+        D_lens = self._lens_cosmo.D_d * 1e6
+
+        r_c = 160 * m22**(-1) * a**(0.5) * m22**(-1) * M9**(-plaw) 
```

Comment:
Can you put the computation of the core radius and the central density into their own respective functions? That way it is 1) easier to write test functions and 2) more readable if someone looks into the code to figure out what's going on. Also, could you keep the functions that compute r_c and the central density as close to what is in the Schive paper as possible, and deal with the unit conversions separately inside the _uldm_args routine?